Tests/StreamChatTests/Repositories/SyncOperations_Tests.swift

```diff
@@ -138,8 +138,8 @@ final class SyncOperations_Tests: XCTestCase {
 
         operation.startAndWaitForCompletion()
 
-        XCTAssertEqual(context.watchedAndSynchedChannelIds.count, 0)
-        XCTAssertNotCall("recoverWatchedChannel(completion:)", on: controller)
+        XCTAssertEqual(context.watchedAndSynchedChannelIds.count, 1)
+        XCTAssertCall("recoverWatchedChannel(completion:)", on: controller)
```

Comment:
Why are we reverting the test here? Although the test passes now, it is not what it is supposed to be testing. Right? @polqf

Reply:
This test should be split in 2.
- One that checks that given a channelId that was synched (present in `context.synchedChannelIds `), it calls `recoverWatchedChannel`, but the end result of `context.watchedAndSynchedChannelIds` is 1
- One that checks that given  a channelId that was NOT synched (NOT present in `context.synchedChannelIds `), it calls `recoverWatchedChannel`, and the result of `context.watchedAndSynchedChannelIds` is 1 and `context.synchedChannelIds` keeps being 0

Reply:
@polqf, the current test was already doing the first check. I've added another one that tests the second point.